include/gridtools/common/int_vector.hpp

```diff
@@ -0,0 +1,179 @@
+                wstd::forward<Vecs>(vecs)...);
+        }
+
+        /**
+         * @brief Returns `int_vector` with elements multiplied by an integral scalar
+         */
+        template <class Vec, class Scalar>
+        GT_FUNCTION GT_CONSTEXPR auto multiply(Vec &&vec, Scalar scalar) {
+            return tuple_util::host_device::transform([scalar](auto v) { return v * scalar; }, wstd::forward<Vec>(vec));
+        }
+
+        /**
+         * @brief Returns `int_vector` with elements removed that are `integral_constant<T, 0>`
+         */
+        template <class Vec>
+        GT_FUNCTION GT_CONSTEXPR auto prune_zeros(Vec &&vec) {
+            using filtered_map_t = meta::filter<meta::not_<impl_::is_constant_zero>::apply, hymap::to_meta_map<Vec>>;
+            using keys_t = meta::transform<meta::first, filtered_map_t>;
+            using generators = meta::transform<impl_::at_key_f, keys_t>;
+            return tuple_util::host_device::generate<generators, hymap::from_meta_map<filtered_map_t>>(
+                wstd::forward<Vec>(vec));
+        }
+
+        namespace arithmetic {
+            template <class Vec,
+                class Scalar,
+                std::enable_if_t<is_int_vector_v<std::decay_t<Vec>> &&
+                                     (std::is_integral_v<Scalar> || is_integral_constant<Scalar>::value),
+                    bool> = true>
+            GT_FUNCTION GT_CONSTEXPR auto operator*(Vec &&vec, Scalar scalar) {
+                return multiply(wstd::forward<Vec>(vec), scalar);
+            }
+
+            template <class Vec,
+                class Scalar,
+                std::enable_if_t<is_int_vector_v<std::decay_t<Vec>> &&
+                                     (std::is_integral_v<Scalar> || is_integral_constant<Scalar>::value),
+                    bool> = true>
+            GT_FUNCTION GT_CONSTEXPR auto operator*(Scalar scalar, Vec &&vec) {
+                return multiply(wstd::forward<Vec>(vec), scalar);
+            }
+
+            template <class First,
+                class Second,
+                std::enable_if_t<is_int_vector_v<std::decay_t<First>> && is_int_vector_v<std::decay_t<Second>>, bool> =
+                    true>
+            GT_FUNCTION GT_CONSTEXPR auto operator+(First &&first, Second &&second) {
+                return plus(wstd::forward<First>(first), wstd::forward<Second>(second));
+            }
+
+            template <class Vec, std::enable_if_t<is_int_vector_v<std::decay_t<Vec>>, bool> = true>
+            GT_FUNCTION GT_CONSTEXPR auto operator+(Vec &&vec) {
+                return vec;
+            }
+
+            template <class Vec, std::enable_if_t<is_int_vector_v<std::decay_t<Vec>>, bool> = true>
+            GT_FUNCTION GT_CONSTEXPR auto operator-(Vec &&vec) {
+                return multiply(wstd::forward<Vec>(vec), integral_constant<int, -1>{}); // TODO not compiling with nvcc
```

Comment:
let us try to resolve TODO. Have you tried to replace `{}` by `()`?

Reply:
I forgot to delete the TODO. It is working in this version